### ISSUE TITLE
Bad pattern

### ISSUE THREAD

**Issue by pjcj** [zdharma/fast-syntax-highlighting#106]
My apologies in advance for a rushed and minimal bug report, but perhaps it's enough for you to recognise and fix a problem.

```
$ echo **/*(.)
-fast-highlight-string-process:23: bad pattern: )
```

Adding emulate zsh at the start of the -fast-highlight-string-process function solves this problem for me.

**Comment by psprint** [comment]
Thanks! I've added a commit 2 days ago, it should fix the issue

**Comment by pjcj** [comment]
Perfect, it works fine for me.  Thanks very much!

**Comment by F-i-f** [comment]
@psprint , the fix you committed breaks f-sy-h on zsh versions lower than 5.x.
The `emulate -o` option was added in 5.x.
I'll submit a PR shortly with a fix.

**Comment by ysmiles** [comment]
meet some problem (emulate: too many arguments).
can verify that F-i-f's fix works for me (zsh 4.3).
